src/figuresClasses.ts

```diff
@@ -1,19 +1,84 @@
-export interface Figure {
-
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
 }
 
-export class Triangle {
+type Color = 'red' | 'green' | 'blue';
+type GetArea = () => number;
 
+export interface Figure {
+  color: Color,
+  shape: Shape,
+  getArea: GetArea,
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public side1: number,
+    public side2: number,
+    public side3: number,
+  ) {
+    if (side1 <= 0 || side2 <= 0 || side2 <= 0) {
+      throw new Error('your error message');
+    }
+
+    const sides: number[] = [this.side1, this.side2, this.side3];
+
+    const max: number = Math.max(...sides);
+
+    const sum = sides.filter((a) => a !== max).reduce((a, b) => a + b);
```

Comment:
simplify it. you can avoid so many actions. just sort it and use indexes for conditional